diff --git framework/filesystem/GD.php framework/filesystem/GD.php
index d34704f..b8d64f0 100644
--- framework/filesystem/GD.php
+++ framework/filesystem/GD.php
@@ -129,7 +129,7 @@ class GDBackend extends Object implements Image_Backend {
 
 		if ($limit < 0) return true; // memory_limit == -1
 
-		$imageInfo = getimagesize($filename);
+		$imageInfo = @getimagesize($filename);
 
 		// bits per channel (rounded up, default to 1)
 		$bits = isset($imageInfo['bits']) ? ($imageInfo['bits'] + 7) / 8 : 1;
diff --git framework/forms/HtmlEditorField.php framework/forms/HtmlEditorField.php
index 4bc987e..4c12319 100644
--- framework/forms/HtmlEditorField.php
+++ framework/forms/HtmlEditorField.php
@@ -94,7 +94,7 @@ class HtmlEditorField extends TextareaField {
 				if($width && $height && ($width != $image->getWidth() || $height != $image->getHeight())) {
 					//Make sure that the resized image actually returns an image:
 					$resized=$image->ResizedImage($width, $height);
-					if($resized) $img->setAttribute('src', $resized->getRelativePath());
+					if($resized) $img->setAttribute('src', $resized->getURL());
 				}
 			}
 
@@ -464,7 +464,7 @@ class HtmlEditorField_Toolbar extends RequestHandler {
 	protected function viewfile_getLocalFileByID($id) {
 		$file = DataObject::get_by_id('File', $id);
 
-		if ($file && $file->canView()) return array($file, $file->RelativeLink());
+		if ($file && $file->canView()) return array($file, $file->getURL());
 		return array(null, null);
 	}
 
diff --git framework/forms/UploadField.php framework/forms/UploadField.php
index 57926dc..7dfefc8 100644
--- framework/forms/UploadField.php
+++ framework/forms/UploadField.php
@@ -870,11 +870,13 @@ class UploadField extends FileField {
 			$width = $this->getPreviewMaxWidth();
 			$height = $this->getPreviewMaxHeight();
 			if ($file->hasMethod('getThumbnail')) {
-				return $file->getThumbnail($width, $height)->getURL();
+               $transform = $file->getThumbnail($width, $height);
+				return $transform ? $transform->getURL() : $file->Icon();
 			} elseif ($file->hasMethod('getThumbnailURL')) {
 				return $file->getThumbnailURL($width, $height);
 			} elseif ($file->hasMethod('Fit')) {
-				return $file->Fit($width, $height)->getURL();
+               $transform = $file->Fit($width, $height);
+               return $transform ? $transform->getURL() : $file->Icon();
 			} else {
 				return $file->Icon();
 			}
diff --git framework/filesystem/File.php framework/filesystem/File.php
index 8d3bfc2..fa6358a 100644
--- framework/filesystem/File.php
+++ framework/filesystem/File.php
@@ -227,7 +227,7 @@ class File extends DataObject {
 	 * @return bool
 	 */
 	public function exists() {
-		return parent::exists() && file_exists($this->getFullPath());
+		return parent::exists(); //  && file_exists($this->getFullPath());
 	}
 
 	/**
@@ -302,7 +302,7 @@ class File extends DataObject {
 		// ensure that the record is synced with the filesystem before deleting
 		$this->updateFilesystem();
 
-		if($this->exists() && !is_dir($this->getFullPath())) {
+		if($this->exists() && !is_dir($this->getFullPath()) && file_exists($this->getFullPath())) {
 			unlink($this->getFullPath());
 		}
 	}
@@ -710,7 +710,9 @@ class File extends DataObject {
 	 * @return string
 	 */
 	public function getURL() {
-		return Controller::join_links(Director::baseURL(), $this->getFilename());
+		$url = Controller::join_links(Director::baseURL(), $this->getFilename());
+		$this->extend('updateURL', $url);
+		return $url;
 	}
 
 	/**
